x/valset-pref/msg_server_test.go

```diff
@@ -642,7 +643,120 @@ func (suite *KeeperTestSuite) TestWithdrawDelegationRewards() {
 
 			} else {
 				suite.Require().Error(err)
+			}
+		})
+	}
+}
 
+func (suite *KeeperTestSuite) TestDelegateBondedTokens() {
+	suite.SetupTest()
+
+	testLock := suite.SetupLocks(sdk.AccAddress([]byte("addr1---------------")))
+	testLock_addr2 := suite.SetupLocks(sdk.AccAddress([]byte("addr2---------------")))
+
+	tests := []struct {
+		name                 string
+		delegator            sdk.AccAddress
+		lockId               uint64
+		expectedUnlockedOsmo sdk.Coin
+		expectedDelegations  []sdk.Dec
+		setValSet            bool
+		expectPass           bool
+	}{
+		{
+			name:                 "DelegateBondedTokens with existing osmo denom lockId, bonded and <= 2 weeks bond duration",
+			delegator:            sdk.AccAddress([]byte("addr1---------------")),
+			lockId:               testLock[0].ID,
+			expectedUnlockedOsmo: sdk.NewCoin(appParams.BaseCoinUnit, sdk.NewInt(60_000_000)), // delegator has 100osmo and creates 5 locks 10osmo each, forceUnlock only 1 lock
+			expectedDelegations:  []sdk.Dec{sdk.NewDec(2_000_000), sdk.NewDec(3_300_000), sdk.NewDec(1_200_000), sdk.NewDec(3_500_000)},
+			setValSet:            true,
+			expectPass:           true,
+		},
+		{
+			name:       "DelegateBondedTokens with existing stake denom lockId, bonded and <= 2 weeks bond duration",
+			delegator:  sdk.AccAddress([]byte("addr9---------------")),
+			lockId:     testLock[1].ID,
+			expectPass: false,
+		},
+		{
+			name:       "DelegateBondedTokens with non existing lockId",
+			delegator:  sdk.AccAddress([]byte("addr3---------------")),
+			lockId:     10,
+			expectPass: false,
+		},
+		{
+			name:       "DelegateBondedTokens with lockOwner != delegatorOwner",
+			delegator:  sdk.AccAddress([]byte("addr4---------------")),
+			lockId:     testLock[2].ID,
+			expectPass: false,
+		},
+		{
+			name:       "DelegateBondedTokens with lock duration > 2 weeks",
+			delegator:  sdk.AccAddress([]byte("addr5---------------")),
```

Comment:
Do we not need to use addr1 as delegator since we used `testLock := suite.SetupLocks(sdk.AccAddress([]byte("addr1---------------")))`? Or else I feel like these cases would fail not because of the description written in the test cases but because of the wrong delegator ( Or maybe Im just missing context) wdyt?

Reply:
It would actually fail because it wouldn't have existing delegations, but now that i think about it it makes sense to keep all the delegator to addr1. And have that delegator attempt to create new locks instead of different delegator creating different locks. 

I'll change all test to use addr1 :))